reference_files/1.0.0/unicode_spp.yaml

```diff
@@ -3,14 +3,14 @@
 %YAML 1.1
 %TAG ! tag:stsci.edu:asdf/
 --- !core/asdf-1.0.0
-asdf_library: !core/software-1.0.0 {author: Space Telescope Science Institute, homepage: 'http://github.com/spacetelescope/asdf',
-  name: asdf, version: 1.0.0}
+asdf_library: !core/software-1.0.0 {author: The ASDF Developers, homepage: 'http://github.com/asdf-format/asdf',
+  name: asdf, version: 3.0.1}
 datatype<U: !core/ndarray-1.0.0
   data: ['', "\U00010020"]
-  datatype: [ucs4, 2]
+  datatype: [ucs4, 1]
```

Comment:
It is unclear to me what `unicode_spp` is supposed to be illustrating. My closest guess is that the character `𐀠` is [U+10020](https://unicode-explorer.com/c/10020) is part of plane 1 the [Supplementary Multilingual Plane](https://en.wikipedia.org/wiki/Plane_(Unicode)#Supplementary_Multilingual_Plane) so perhaps this was suppored to be `smp` not `spp`. On my system this is stored in 4 byte unicode which explains the `[ucs4, 1]` change in this file. I'm not sure what system or version of numpy was using 8 bytes to store this character.
```python
>> np.unicode_('𐀠').dtype.itemsize
4
```
The `generate.py` file could be updated to use `U2` instead of `U` to generate a file that uses 8 bytes:
```python
>> np.array(['𐀠'], dtype='U2').dtype.itemsize
8
```
At this point I suggest we hold off making this change to `generate.py` and instead include the updated file in this PR as it offers a slightly different example for unicode in ASDF compared to the `bmp` file (which has a combined character which results in `[ucs4, 2]`).